`TypeError` occurs during handling of `CUDADriverError` in parameterize test

An error is instantiated to reraise the error here.
https://github.com/cupy/cupy/blob/818008578179a8f8299175691634f497dccb2721/cupy/testing/parameterized.py#L92
However, `cupy.cuda.driver.CUDADriverError` does not support an interface of this format.
https://github.com/cupy/cupy/blob/818008578179a8f8299175691634f497dccb2721/cupy/cuda/driver.pyx#L105-L108
So, when a parameterized test was failed due to `CUDADriverError`, the error is not re-raised correctly.
```py
During handling of the above exception, another exception occurred:
cupy/testing/parameterized.py:92: in new_method
    raise e.__class__(s.getvalue()).with_traceback(e.__traceback__)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   def __init__(self, Result status):
E   TypeError: an integer is required

cupy/cuda/driver.pyx:107: TypeError
```

--- a/cupy/testing/parameterized.py
+++ b/cupy/testing/parameterized.py
@@ -6,6 +6,7 @@ import typing as tp  # NOQA
 import unittest
 
 from cupy.testing import _bundle
+from cupy.cuda import driver
 
 
 def _param_to_str(obj):
@@ -89,7 +90,10 @@ def _parameterize_test_case(base, i, param):
                 s.write('Test parameters:\n')
                 for k, v in sorted(param.items()):
                     s.write('  {}: {}\n'.format(k, v))
-                raise e.__class__(s.getvalue()).with_traceback(e.__traceback__)
+                err_class = e.__class__
+                if isinstance(e, driver.CUDADriverError):
+                    err_class, = err_class.__bases__
+                raise err_class(s.getvalue()).with_traceback(e.__traceback__)
         return new_method
 
     return (cls_name, mb, method_generator)

